### PR TITLE
Add azure storage

### PR DESCRIPTION
**Pull request by nitang22**

### CODE REVIEW

File: packages/azure-storage/files/helm-chart/templates/NFS.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/part-of: {{ .Values.organization }}.{{ .Values.project }}
+    app.kubernetes.io/managed-by: kubepak
+  name: {{ .Values.organization }}{{ .Values.project }}nfs
+spec:
+  forProvider:
+    accountKind: "BlockBlobStorage"
+    accountTier: "Premium"
+    accountReplicationType: ZRS
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/NFS.yaml:15]: we can move to `GRS` or `GZRS` in production. During our testing, we can use a simple `LRS`
```suggestion
    accountReplicationType: "{{ index .Values "packages" "azure-storage" "nfs" "accountReplicationType"}}"
```

File: packages/azure-storage/azure-storage.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+hook_initialize() {
+    k8s_namespace_create "${K8S_PACKAGE_NAMESPACE}"
+
+    registry_credentials_add_namespace "${K8S_PACKAGE_NAMESPACE}"
+}
+
+hook_install() {
+    package_helm_install "${K8S_PACKAGE_NAME}" "${K8S_PACKAGE_NAMESPACE}" "${PACKAGE_DIR}/files/helm-chart"
+
```
Comment (fcarregen) [inline packages/azure-storage/azure-storage.sh:48]: do we need to wait for the creation to complete like in MySQL/PostgreSQL?

File: packages/azure-storage/files/helm-chart/templates/NFS.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    allowNestedItemsToBePublic: false
+    infrastructureEncryptionEnabled: true
+    isHnsEnabled: true
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/NFS.yaml:18]: Do we need all the time _hierarchical namespace_?
```suggestion
    isHnsEnabled: "{{ index .Values "packages" "azure-storage" "nfs" "hnsEnabled"}}"
```

File: packages/azure-storage/files/helm-chart/templates/Blobs.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/part-of: {{ .Values.organization }}.{{ .Values.project }}
+    app.kubernetes.io/managed-by: kubepak
+  name: {{ .Values.organization }}{{ .Values.project }}
+spec:
+  forProvider:
+    accountKind: "StorageV2"
+    accessTier: "Hot"
+    accountTier: "Standard"
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/Blobs.yaml:15]:
```suggestion
    accountTier: "{{ index .Values "packages" "azure-storage" "blob" "accountTier"}}"
```

File: packages/azure-storage/files/helm-chart/templates/Blobs.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  labels:
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/part-of: {{ .Values.organization }}.{{ .Values.project }}
+    app.kubernetes.io/managed-by: kubepak
+  name: {{ .Values.organization }}{{ .Values.project }}
+spec:
+  forProvider:
+    accountKind: "StorageV2"
+    accessTier: "Hot"
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/Blobs.yaml:14]:
```suggestion
    accessTier: "{{ index .Values "packages" "azure-storage" "blob" "accessTier"}}"
```

File: packages/azure-storage/files/helm-chart/templates/Blobs.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    accountTier: "Standard"
+    accountReplicationType: ZRS
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/Blobs.yaml:16]:
```suggestion
    accountReplicationType: "{{ index .Values "packages" "azure-storage" "blob" "accountReplicationType"}}"
```

File: packages/azure-storage/files/helm-chart/templates/Blobs.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    allowNestedItemsToBePublic: false
+    # identity:
+    #   - identityIds:
+    #     type:
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/Blobs.yaml:20]: Must be implemented as an `if` block?

File: packages/azure-storage/files/helm-chart/templates/Blobs.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    infrastructureEncryptionEnabled: true
+    largeFileShareEnabled: true
+    location: "East US"
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/Blobs.yaml:23]:
```suggestion
    location: "{{ index .Values "packages" "azure-storage" "location"}}"
```

File: packages/azure-storage/files/helm-chart/templates/Blobs.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    resourceGroupName: "poc-crossplane"
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/Blobs.yaml:24]:
```suggestion
    resourceGroupName: "{{ index .Values "packages" "azure-storage" "resourceGroupName"}}"
```

File: packages/azure-storage/files/helm-chart/templates/Blobs.yaml [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  providerConfigRef:
+    name: crossplane-azure-provider
+  writeConnectionSecretToRef:
+    name: bucket-details
+    namespace: {{ .Release.Namespace }}
+    # tags:
+    #   - key: "AHHH"
+    #     value: "Wooooo"
```
Comment (fcarregen) [inline packages/azure-storage/files/helm-chart/templates/Blobs.yaml:32]: Must be an `if` block

File: packages/azure-storage/azure-storage.sh [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+#
+
+set -eo pipefail
+
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/../../support/scripts/package.sh"
+
+#-----------------------------------------------------------------------------
+# Package Options
+
+# @package-option attributes="final"
```
Comment (fcarregen) [inline packages/azure-storage/azure-storage.sh:27]: If it's like databases, it can't be `final`

**Review by fcarregen — CHANGES_REQUESTED** [review]

File: packages/azure-storage/files/helm-chart/templates/NFS.yaml [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  labels:
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/part-of: {{ .Values.organization }}.{{ .Values.project }}
+    app.kubernetes.io/managed-by: kubepak
+  name: {{ .Values.organization }}{{ .Values.project }}nfs
+spec:
+  forProvider:
+    accountKind: "BlockBlobStorage"
+    accountTier: "Premium"
```
[the remark this answers is not in the document]
Comment (nitang22) [reply packages/azure-storage/files/helm-chart/templates/NFS.yaml:14]: Yes

File: packages/azure-storage/files/helm-chart/templates/NFS.yaml [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    infrastructureEncryptionEnabled: true
+    isHnsEnabled: true
+    location: "East US"
+    networkRules:
+      - defaultAction: "Deny"
+        # Need to whitelist your ip if you want to deploy from your local cluster
+        # ipRules:
+        #   - 107.171.249.3
+        virtualNetworkSubnetIds:
+          - "{{ index .Values "packages" "azure-storage" "forProvider" "networkRules" "virtualNetworkSubnetIds"}}"
```
[the remark this answers is not in the document]
Comment (nitang22) [reply packages/azure-storage/files/helm-chart/templates/NFS.yaml:26]: To be consistent with the configuration files

File: packages/azure-storage/files/helm-chart/templates/NFS.yaml [hunk 14]
```diff
@@ -0,0 +1,47 @@
+---
+apiVersion: storage.azure.upbound.io/v1beta1
+kind: Account
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/part-of: {{ .Values.organization }}.{{ .Values.project }}
+    app.kubernetes.io/managed-by: kubepak
+  name: {{ .Values.organization }}{{ .Values.project }}nfs
```
[the remark this answers is not in the document]
Comment (nitang22) [reply packages/azure-storage/files/helm-chart/templates/NFS.yaml:10]: YES !

**Review by fcarregen — APPROVED** [review]

**Pull request merged**